include/utility/vpd_specific_utility.hpp

```diff
@@ -355,6 +355,12 @@ inline void getVpdDataInVector(const std::string& vpdFilePath,
 {
     try
     {
+        if (std::filesystem::file_size(vpdFilePath) == 0)
```

Comment:
Why is this call required explicitly when file size is already being fetched?

Reply:
calling parse() API directly from Parser object with empty input file, seeing crash in this API by unit test case.
So added this initial check for the file before start reading this file.

Reply:
But that is already being done, why can't you use that?